src/submodules/io/jld2_integration.jl

```diff
@@ -80,42 +91,63 @@ struct SerializableAStar{D,P,M}
     cost_metric::Pathfinding.CostMetric{D}
 end
 
-to_serializable(t::ABM) = SerializableABM(
-    collect(values(t.agents)),
-    to_serializable(t.space),
-    to_serializable(t.properties),
-    t.rng,
-    t.maxid.x,
-)
+function to_serializable(t::ABM{S}) where {S}
```

Comment:
A lot of this section should use helper functions from [here](https://github.com/JuliaDynamics/Agents.jl/blob/22076028a00cc24ac619567627967bcd76c97268/src/core/model.jl#L218-L230).

`nagents(model)` vs `length(sabm.agents)`
`allagents(model)` vs `values(t.agents)`

etc.